examples/filetable.py

```diff
@@ -15,80 +16,96 @@ def filetable_to_firefly(ffclient, topdir, pattern, recursive=True):
+                                 "file:///external" + path])
 
     tbl_val = ffclient.upload_file(fd.name)
     os.remove(fd.name)
-    return(tbl_val, metadict)
+    return (tbl_val, metadict)
 
 
 # Sample application
 def main():
     import argparse
-    parser = argparse.ArgumentParser(description="""
+
+    parser = argparse.ArgumentParser(
+        description="""
         Display a table of files in a Firefly window
-        """)
-    parser.add_argument('topdir', help='top-level directory to search')
-    parser.add_argument('pattern', help='filename pattern for search')
-    parser.add_argument('--norecursion', help='do not recursively search topdir',
-                        action='store_true')
-    parser.add_argument('--host', help='host address for Firefly',
-                        default='localhost')
-    parser.add_argument('--port', help='port for Firefly', default='8080')
-    parser.add_argument('--base', help='base.for Firefly', default='firefly')
-    parser.add_argument('--channel', help='channel name for websocket',
-                        default=None)
-    parser.add_argument('--printurl', help='print browser url instead of' +
-                        ' attempting to launch browser', action='store_true')
+        """
+    )
+    parser.add_argument("topdir", help="top-level directory to search")
+    parser.add_argument("pattern", help="filename pattern for search")
+    parser.add_argument(
+        "--norecursion", help="do not recursively search topdir", action="store_true"
+    )
+    parser.add_argument(
+        "--firefly_url", help="URL for Firefly server", default=os.getenv("FIREFLY_URL")
+    )
+    parser.add_argument("--channel", help="channel name for websocket", default=None)
+    parser.add_argument(
+        "--printurl",
+        help="print browser url instead of" + " attempting to launch browser",
+        action="store_true",
+    )
 
     args = parser.parse_args()
     topdir = args.topdir
     pattern = args.pattern
-    host = args.host
-    port = args.port
-    basedir = args.base
+    firefly_url = args.firefly_url
     channel = args.channel
     printurl = args.printurl
+    launch_browser = True
+    if printurl:
+        launch_browser = False
     recursion = not args.norecursion
 
-    fc = FireflyClient.make_client(url='{}:{}'.format(host, port), channel_override=channel, html_file='slate.html')
+    fc = firefly_client.FireflyClient.make_client(
+        url=firefly_url,
+        channel_override=channel,
+        html_file="slate.html",
+        launch_browser=launch_browser,
+    )
+    if printurl:
+        print("Firefly URL is {}".format(fc.get_firefly_url()))
+
+    print("Searching for files...", end="")
+    tbl_val, metainfo = filetable_to_firefly(fc, topdir, pattern, recursive=recursion)
+    print("done.")
+
     if printurl:
-        print('Firefly url is {}'.format(fc.get_firefly_url()))
+        input("Press Enter after your browser is open to the Firefly URL...")
```

Comment:
I was not immediately clear to me that it was referring to the printed URL.
```suggestion
        input("Press Enter after you have opened the Firefly URL printed above")
```